app/controllers/files/browse_controller.rb

```diff
@@ -10,4 +11,28 @@ def browse
     # load files
     @file = @identity.browse(path)
   end
+
+
+  def upload
+    uploader = LocalFileUploader.new
+
+    uploader.store!(params[:bruse_file][:file])
+
+    file = BruseFile.new
+
+    file.foreign_ref = uploader.file.file
```

Comment:
I'm not sure if I think it's a good idea to store the absolute file path (for example `/Users/kalas/dev/bruse/usercontent/56ad62e1-ffc7-4...`), like we do here. I think this leads to more problems than storing the file *name* only (for example `56ad62e1-ffc7-4...`). Or am I missing something here?

If we store the path, we risk exposing potential sensitive system information, files can't be moved. Also, we break the relative path structure that our rails app has. (Meaning that our app can be moved anywhere without risk.)